subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -112,7 +113,13 @@ static int cracen_signature_prepare_ec_prvkey(struct si_sig_privkey *privkey, ch
 		return status;
 	}
 
-	if (key_buffer_size != PSA_BITS_TO_BYTES(psa_get_key_bits(attributes))) {
+	if (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
+	    PSA_KEY_LOCATION_CRACEN_KMU) {
+		key_buffer = kmu_push_area;
+		key_buffer_size = sizeof(kmu_push_area);
```

Comment:
I think this should be actual key size, not kmu push area size(64 bytes), i guess just using that value from the attributes should be fine.

Reply:
That was the other option I had in mind but in case the bits are not getting checked, the buffer size could be set to a value that is bigger than the buffer.